lib/wrappers/transform-node.js

```diff
@@ -21,7 +26,35 @@ module.exports = class TransformNodeWrapper extends NodeWrapper {
     return new Promise(resolve => {
       startTime = process.hrtime();
 
-      if (!this.nodeInfo.persistentOutput) {
+      // TODO: "sideEffectFree" needs to be added to broccoli/broccoli-node-info for this to 
+      // work without overriding __broccoliGetInfo__ at the plugin layer
+      if (this.nodeInfo.sideEffectFree) {
+        let shouldBuild = false;
```

Comment:
some `debug` logging to let us know what is sideeffectfree and what isn't would be nice to have.

Reply:
Done in latest commit